openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -186,12 +186,20 @@ def update_account_settings(requesting_user, update, username=None):
     additional_fields = {"sch_org", "phone", "organization_type"}.intersection(set(update.keys()))
     if additional_fields:
         instance = AdditionalRegistrationFields.objects.filter(user=existing_user).first()
+        data = {}
+        AdditionalRegistrationFields.validate_data(update, additional_fields, data, field_errors)
```

Comment:
We ourselves are implementing the `validate_data` method right? can we change the implementation to return validated data instead of updated an argument? I think we should try to keep methods as pure as possible